src/tooltip.tsx

```diff
@@ -178,84 +249,146 @@ const Tooltip: React.FC<Props> = ({
-                    targetBoundingClientRect.current.height / 2,
-            },
-            left: {
-                left: targetBoundingClientRect.current.left - width - distanceToTarget,
-                top:
-                    window.pageYOffset +
-                    targetBoundingClientRect.current.top +
-                    targetBoundingClientRect.current.height / 2,
-            },
-            top: {
-                top: window.pageYOffset + targetBoundingClientRect.current.top - distanceToTarget,
-                left: isTabletOrSmaller
-                    ? marginLeftRightMobile
-                    : window.pageXOffset +
-                      targetBoundingClientRect.current.left +
-                      targetBoundingClientRect.current.width / 2 -
-                      width / 2,
-            },
-            bottom: {
-                top: window.pageYOffset + targetBoundingClientRect.current.bottom + distanceToTarget,
-                left: isTabletOrSmaller
-                    ? marginLeftRightMobile
-                    : window.pageXOffset +
-                      targetBoundingClientRect.current.left +
-                      targetBoundingClientRect.current.width / 2 -
-                      width / 2,
-            },
-        };
-        return containerPos[position];
-    };
+        const arrowPosition =
+            tooltipClient.width > targetBoundingClientRect.current.width &&
+            targetBoundingClientRect.current.left + targetBoundingClientRect.current.width / 2 <
+                tooltipClient.width / 2 + 16
+                ? targetBoundingClientRect.current.width / 2
+                : '50%';
+        const aux = position === 'bottom' || position === 'top' ? {left: arrowPosition} : {};
 
-    const getCustomStylesForMobile = () =>
-        isTabletOrSmaller
-            ? {
-                  left:
-                      targetBoundingClientRect.current.left +
-                      targetBoundingClientRect.current.width / 2 -
-                      marginLeftRightMobile,
-              }
-            : {};
+        return aux;
+    };
 
-    const getWidth = () =>
-        isTabletOrSmaller ? window.innerWidth - marginLeftRightMobile * 2 : getWidthDesktop(rest.width);
+    const getContainerPosition = React.useCallback(
+        (position: Position, width: number) => {
+            if (isServerSide() || !tooltipClient) {
+                return {};
+            }
+
+            const tooltipMarginFix =
+                tooltipClient.width > targetBoundingClientRect.current.width &&
+                targetBoundingClientRect.current.left + targetBoundingClientRect.current.width / 2 <
+                    tooltipClient.width / 2 + 16
+                    ? Math.round(tooltipClient.width / 2 - targetBoundingClientRect.current.width / 2)
+                    : 0;
+
+            const bottomAdjustment = changedPosition
+                ? changedPosition
+                : !width
```

Comment:
what's your intention here? `bottomAdjustment` is meant to be a number? then why are you assigning string values here? (isnt `changedPosition` a string?) this doesn't make any sense to me